src/proj_capi.jl

```diff
@@ -0,0 +1,163 @@
+# The following functions are generally named after the associated C API
+# functions, but without the pj prefix.
+
+immutable ProjUV
+    u::Cdouble
+    v::Cdouble
+end
+
+@doc "forward projection from Lat/Lon to X/Y (only supports 2 dimensions)" ->
+function _fwd!(lonlat::Vector{Cdouble}, proj_ptr::Ptr{Void})
+    xy = ccall((:pj_fwd, libproj), ProjUV, (ProjUV, Ptr{Void}), ProjUV(lonlat[1], lonlat[2]), proj_ptr)
+    _errno() != 0 &&  error("forward projection error: $(_strerrno())")
```

Comment:
Stylistic nitpick: I find this really hard to read.  The following "assertion style" equivalent works much better for me, since it can be read as a valid English sentence:

```julia
_errno() == 0 || error("forward projection error: $(_strerrno())")
```